lgi/marshal.c

```diff
@@ -68,16 +83,21 @@ marshal_2c_int (lua_State *L, GITypeTag tag, GIArgument *val, int narg,
 	  }								\
 	break
 
-      HANDLE_INT(INT8, int8, INT, gint, -0x80, 0x7f, s);
-      HANDLE_INT(UINT8, uint8, UINT, guint, 0, 0xff, u);
-      HANDLE_INT(INT16, int16, INT, gint, -0x8000, 0x7fff, s);
-      HANDLE_INT(UINT16, uint16, UINT, guint, 0, 0xffff, u);
-      HANDLE_INT(INT32, int32, INT, gint, -0x80000000LL, 0x7fffffffLL, s);
-      HANDLE_INT(UINT32, uint32, UINT, guint, 0, 0xffffffffUL, u);
-      HANDLE_INT(UNICHAR, uint32, UINT, guint, 0, 0x7fffffffLL, u);
+      HANDLE_INT(INT8, int8, INT, gint, G_MININT8, G_MAXINT8, s);
+      HANDLE_INT(UINT8, uint8, UINT, guint, 0, G_MAXUINT8, u);
+      HANDLE_INT(INT16, int16, INT, gint, G_MININT16, G_MAXINT16, s);
+      HANDLE_INT(UINT16, uint16, UINT, guint, 0, G_MAXUINT16, u);
+      HANDLE_INT(INT32, int32, INT, gint, G_MININT32, G_MAXINT32, s);
+      HANDLE_INT(UINT32, uint32, UINT, guint, 0, G_MAXUINT32, u);
+      HANDLE_INT(UNICHAR, uint32, UINT, guint, 0, G_MAXUINT32, u);
```

Comment:
Why are you replacing the constants with glib macros here (and not in the `#else` below)? That seems like an unrelated change. Or was there something wrong with them?

Reply:
I was changing this part of code anyways, so was thinking that hardcoding stuff is generally bad. Result works flawlessly.

As for the `#else` clause, I left those because I'm not sure about pre-integers 64-bits stuff.